### PR TITLE
ScriptInfo versioning improvements

### PR DESCRIPTION
**Pull request by sheetalkamat**
- aeec611d7c530c55b9d03c98dffc7e6c231f744c With this change, if file is opened, it is not immediately switched to versioning cache (which is used for editing). It is deferred till we actually receive edit request or its opened with different contents than on the disk.
- 327675c25b65517fdf1b86939d566519f49a16ca When closing file, it does not read contents of file right away. Instead defers it till next project update.

The other commits are test changes to baseline file version and text for program in the baseline done to ensure the changes dont unexpectedly change versioning and text in the program.

@andrewbranch discovered this while investigating #52792 though as discussed offline it doesnt fix that issue completely.

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
Can you merge and update baselines?

**Comment by andrewbranch** [comment]
@typescript-bot pack this

File: tests/baselines/reference/api/tsserverlibrary.d.ts [hunk 1]
```diff
@@ -3081,10 +3077,9 @@ declare namespace ts {
             readonly containingProjects: Project[];
             private formatSettings;
             private preferences;
-            private textStorage;
-            constructor(host: ServerHost, fileName: NormalizedPath, scriptKind: ScriptKind, hasMixedContent: boolean, path: Path, initialVersion?: ScriptInfoVersion);
+            constructor(host: ServerHost, fileName: NormalizedPath, scriptKind: ScriptKind, hasMixedContent: boolean, path: Path, initialVersion?: number);
```
Comment (jakebailey) [inline tests/baselines/reference/api/tsserverlibrary.d.ts:3080]: Given this breaking change, is anyone depending on this taht we should worry about? Is there a way to write this code as to not break existing users, i.e. keep the old thing around, deprecate, and pull a version out of it?

Comment (sheetalkamat) [reply tests/baselines/reference/api/tsserverlibrary.d.ts:3080]: I have looked up but didn't find any usage for this which is why i just modified this without adding overload.
You think overload thats marked deprecated is better even then?

Comment (jakebailey) [reply tests/baselines/reference/api/tsserverlibrary.d.ts:3080]: Yeah, now that I'm looking, I've only been able to find 3 examples of this being used (a subset of https://sourcegraph.com/search?q=context:global+/new+.*%5CbScriptInfo%5C%28/+lang:typescript+-file:harness+-file:editorServices+-file:languageServiceHost&patternType=standard&sm=1&groupBy=repo are the right thing), and those all a decade old and unmaintained. So, probably safe.

**Review by jakebailey — APPROVED** [review]
So, code wise, I think this looks correct. Just so I understand it, we're effectively removing the double-versioning and now have one version number, consulting `ScriptVersionCache` only when present, and more lazily creating it?

But @andrewbranch probably will have a better idea of what the intent was.

**Review by andrewbranch — APPROVED** [review]
Awesome, thank you! I’ll follow up on the remaining case where the export map cache needs to be cleared.

**Pull request merged**